exp/api/v1alpha3/machinepool_types.go

```diff
@@ -59,6 +59,9 @@ type MachinePoolSpec struct {
 	// This field must match the provider IDs as seen on the node objects corresponding to a machine pool's machine instances.
 	// +optional
 	ProviderIDList []string `json:"providerIDList,omitempty"`
+
+	// FailureDomains is the list of failure domains this MachinePool should be attached to.
+	FailureDomains clusterv1.FailureDomains `json:"failureDomains,omitempty"`
```

Comment:
I'm wondering if it would make sense to use a []string or []*string here rather than using the `clusterv1.FailureDomains` type here, I wouldn't expect the underlying implementation to need the additional information from the FailureDomains type. It would also more closely match the types/usage with how it is currently being used in Machine* types today.

Reply:
Changed to be `[]string`.